tools/api/generate_go_protobuf.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python3
+from subprocess import check_output
+from subprocess import check_call
+
+import glob
+import os
+import shutil
+import sys
+import re
+
+targets = '@envoy_api//...'
+import_base = 'github.com/envoyproxy/go-control-plane'
+output_base = 'build_go'
+repo_base = 'go-control-plane'
+branch = 'master'
+mirror_msg = 'Mirrored from envoyproxy/envoy @ '
+user_name = 'Kuat Yessenov'
+user_email = 'kuat@google.com'
+
+
+def generateProtobufs(output):
+  bazel_bin = check_output(['bazel', 'info', 'bazel-bin']).decode().strip()
+  go_protos = check_output([
+      'bazel',
+      'query',
+      'kind("go_proto_library", %s)' % targets,
+  ]).split()
+
+  # Each rule has the form @envoy_api//foo/bar:baz_go_proto.
+  # First build all the rules to ensure we have the output files.
+  check_call(['bazel', 'build', '-c', 'fastbuild'] + go_protos)
+
+  shutil.rmtree(output, ignore_errors=True)
+  for rule in go_protos:
+    # Example rule:
+    # @envoy_api//envoy/config/bootstrap/v2:pkg_go_proto
+    #
+    # Example generated directory:
+    # bazel-bin/external/envoy_api/envoy/config/bootstrap/v2/linux_amd64_stripped/pkg_go_proto%/github.com/envoyproxy/go-control-plane/envoy/config/bootstrap/v2/
+    #
+    # Example output directory:
+    # go_out/envoy/config/bootstrap/v2
+    rule_dir, proto = rule.decode()[len('@envoy_api//'):].rsplit(':', 1)
+
+    input_dir = os.path.join(bazel_bin, 'external', 'envoy_api', rule_dir, 'linux_amd64_stripped',
+                             proto + '%', import_base, rule_dir)
+    input_files = glob.glob(os.path.join(input_dir, '*.go'))
+    output_dir = os.path.join(output, rule_dir)
+
+    # Ensure the output directory exists
+    os.makedirs(output_dir, 0o755, exist_ok=True)
+    for generated_file in input_files:
+      shutil.copy(generated_file, output_dir)
+  print('Go artifacts placed into: ' + output)
+
+
+def cloneGoProtobufs(repo):
+  # Create a local clone of go-control-plane
+  shutil.rmtree(repo, ignore_errors=True)
+  check_call(['git', 'clone', 'git@git:envoyproxy/go-control-plane', repo])
```

Comment:
Nit: I'd refactor these to have a distinct `git` method, with optional repo. I.e. just remove some of the boilerplate.

Reply:
Ping on this one, I think we should at least clean this boiler plate up if we're not going to switch to pygithub.

Reply:
Done that.